src/linux/minidump_writer.rs

```diff
@@ -154,16 +175,15 @@ impl MinidumpWriter {
             }
 
             if !self.crash_thread_references_principal_mapping(&dumper) {
-                return Err(InitError::PrincipalMappingNotReferenced.into());
+                soft_errors.push(WriterError::PrincipalMappingNotReferenced);
             }
         }
 
         let mut buffer = Buffer::with_capacity(0);
-        self.generate_dump(&mut buffer, &mut dumper, destination)?;
+        self.generate_dump(&mut buffer, &mut dumper, soft_errors, destination)?;
 
-        // dumper would resume threads in drop() automatically,
-        // but in case there is an error, we want to catch it
-        dumper.resume_threads()?;
+        // TODO - Record these errors? Or maybe we don't care?
+        dumper.resume_threads(error_graph::strategy::DontCare);
```

Comment:
Ah sorry, it's not very obvious why this comment is in here, but the reason this is non-trivial is because the minidump has already been written by this point. I guess I would have to move this into `generate_dump()` after access to the thread information is no longer needed. I can do it, I just wanted to check with you that I should first?

Reply:
Ah right, I hadn't thought about that. I was still thinking about the old scheme where we returned an error to the caller. Don't bother then, it's not very important.

Reply:
Ah dang it -- I already did it :laughing: